pyrokinetics/gk_code/GKInputGS2.py

```diff
@@ -549,3 +559,14 @@ def set(
 
         for name, namelist in self.data.items():
             self.data[name] = convert_dict(namelist, local_norm.gs2)
+
+    def get_ne_te_normalisation(self):
+        # Load each species into a dictionary
+        for i_sp in range(self.data["species_knobs"]["nspec"]):
+            gs2_key = f"species_parameters_{i_sp + 1}"
+            if self.data[gs2_key]["z"] == -1:
```

Comment:
In GS2 we are free to set the normalising charge so this won't necessarily always find the electrons.

Reply:
Oh god why would someone do that.

I guess we can use the species type instead?

Reply:
The only reasons I've come across so far are:
1. Testing (check choice of normalisation changes result in expected way).
2. To allow one to change the ion and electron direction convention (I'd guess at least, picking q_electron = 1 might give ion frequencies as negative, matching other codes).

Unfortunately the species type doesn't promise this either really as it only influences the collision operator and is probably really meant to mean "light species". I think it's not unreasonable to use this though. You might be able to do something like, check if there are species with both signs of charge, if so take the one with the smallest mass as the electrons. If only one sign of charge then either fall back to type/adiabatic_option if only one kinetic specie or assume adiabatic electron if >1 kinetic specie? All getting quite complicated so perhaps pyro should just make a clear assumption.

Reply:
Ive made a check for species and type and raise an error if the electron isnt found